XenModel/XenSearch/GroupingTypes.cs

```diff
@@ -277,7 +277,7 @@ public override String GroupingName
 
         public override String GetGroupName(Object group)
         {
-            if (!(group is T) || i18ns == null)
+            if (!(group is T) || i18ns == null || !i18ns.ContainsKey((T)group))

```

Comment:
Was this just added as a precaution, or was there a case where it caused an exception? I would rather make sure we don't get in here if this condition is violated, and redesign the calling function if necessary. Maybe an Assert would be appropriate?